src/utils.ts

```diff
@@ -173,15 +173,27 @@ export function areMapsEqual(
   let isValueEqual = a.size === b.size;
 
   if (isValueEqual && a.size) {
+    const matchedIndices: Record<number, true> = {};
+
     a.forEach((aValue, aKey) => {
```

Comment:
I have a suggestion that is unrelated to the fix : we could exit early of loops to improve performance. We could use `for of` with the Map or Set as they are iterable and break out of loop b (the second one) when a match is found and break out of loop a (the first one) when no match was found from the last loop (before setting hasMatch to false).

Reply:
Ah yes, the reason for using `forEach` is namely out of IE11 support (which has `forEach` support but not `for...or` due to lack of `Symbol` support without polyfills). Now that IE11 is no longer supported, it may be time to re-examine this (or have `for...of` as primary, falling back to `forEach` when not supported), but it is outside the scope of this PR.

Reply:
Right, I forgot about IE11 👍
Out of place and general question : where would be a good place for out-of-scope comments? In an issue created for it ? A comment in the PR instead of the review? I don't know and I realize it clutters unnecessarily the review.

Reply:
Yup, fair question. I haven't had to do it a ton prior, but I'm trying to get more structured with requests like this, so its a good a time as any to create a convention.

I think creating an issue is appropriate for these scenarios. I can add the corresponding label for it (question, enhancement, investigation, idea) and prioritize accordingly.